File: Go/Pascal's Triangle II.go
// 0ms, 100.00%; 2MB, 100.00%
func getRow(rowIndex int) []int {
	row := []int{1}
	for i := 1; i <= rowIndex; i++ {
		newRow := make([]int, i+1, i+1)
		newRow[0], newRow[i] = 1, 1
		for j := 1; j < i; j++ {
			newRow[j] = row[j-1] + row[j]
		}
		row = newRow
	}
	return row
}

File: Java/Pascal's Triangle II.java
// 1ms, 76.65%; 37MB, 6.17%
class Solution {
    public static List<Integer> getRow(int rowIndex) {
        List<Integer> res = Arrays.asList(1);
        for (int i = 0; i < rowIndex; ++i) {
            List<Integer> tmp = new ArrayList<>(Arrays.asList(1));
            for (int j = 1; j <= i; ++j) {
                tmp.add(res.get(j - 1) + res.get(j));
            }
            tmp.add(1);
            res = tmp;
        }
        return res;
    }
}
